components/browser/menu/src/main/java/mozilla/components/browser/menu/item/BrowserMenuCheckbox.kt

```diff
@@ -14,9 +20,36 @@ import mozilla.components.browser.menu.R
  * @param listener Callback to be invoked when this menu item is checked.
  */
 class BrowserMenuCheckbox(
-    label: String,
-    initialState: () -> Boolean = { false },
-    listener: (Boolean) -> Unit
-) : BrowserMenuCompoundButton(label, initialState, listener) {
+    private val label: String,
+    private val initialState: () -> Boolean = { false },
+    private val listener: (Boolean) -> Unit
+) : BrowserMenuItem {
```

Comment:
Can you help me understand why we copy (and change?) the implementation here and can't rely on `BrowserMenuCompoundButton` anymore?

CC @jonalmeida, who I think created this base class.

Reply:
`BrowserMenuCheckbox` is now a `LinearLayout` at the parent level instead of an `AppCompatCheckbox`, because of this it cannot be cast `BrowserMenuCompoundButton` without crashing. We also want to add custom `onClick` behavior.

Reply:
Sorry, I may have missed something, I don't follow why compound buttons need to have a LinearLayout to alter the width. Couldn't we just change the `layout_width` in `mozac_browser_menu_item_switch` and `mozac_browser_menu_item_checkbox`?

Reply:
@jonalmeida unfortunately, no.

@NotWoods and I tried to do this, but it seems there's a bug with the way we were using `AppCompatCheckbox` that made it so doing `layout_width` was not respected (the text would get clipped). I believe this is because we were using `drawableEnd` and setting the actual button itself to `null` which is a bit of a hacky solution to begin with.